frontend/src/components/Layout/Carousel/CarouselProgressBar.vue

```diff
@@ -4,147 +4,63 @@
       v-for="i in pages"
       :key="`progress-key-${i}`"
       :class="useResponsiveClasses('progress-bar')"
-      @click.capture="onProgressClicked(i)">
+      @click.capture="emit('progressClicked', i - 1)">
       <div
-        ref="progress"
-        :class="
-          expand
-            ? useResponsiveClasses(
-                'progress d-flex align-center justify-center expand'
-              )
-            : useResponsiveClasses(
-                'progress d-flex align-center justify-center'
-              )
-        " />
+        :class="useResponsiveClasses(barClasses[i - 1])"
+        @animationend="emit('animationEnd')" />
     </div>
   </div>
 </template>
 
-<script lang="ts">
-import { defineComponent } from 'vue';
+<script setup lang="ts">
+import { computed } from 'vue';
+import { useDisplay } from 'vuetify';
 import { useResponsiveClasses } from '@/composables';
 
-export default defineComponent({
-  props: {
-    pages: {
-      type: Number,
-      required: true
-    },
-    currentIndex: {
-      type: Number,
-      required: true
-    },
-    duration: {
-      type: Number,
-      required: true
-    },
-    paused: {
-      type: Boolean,
-      required: true,
-      default: false
-    },
-    hoverable: {
-      type: Boolean,
-      required: false,
-      default: false
+const props = withDefaults(
+  defineProps<{
+    pages: number;
+    currentIndex: number;
+    duration: number;
+    paused: boolean;
+    hoverable?: boolean;
+  }>(),
+  { hoverable: false }
+);
+
+const emit = defineEmits<{
+  (e: 'animationEnd'): void;
+  (e: 'progressClicked', index: number): void;
+}>();
+const display = useDisplay();
+const expand = computed(() => props.hoverable && !display.mobile.value);
+const animDuration = computed(() => (props.duration / 1000).toString() + 's');
+const defaultBarClasses = Object.freeze([
+  'progress',
+  'd-flex',
+  'align-center',
+  'justify-center'
+]);
+const barClasses = computed(() => {
+  return Array.from({ length: props.pages }).map((_, i) => {
```

Comment:
You can directly return the Array.from instead

```suggestion
const barClasses = computed(() =>
  Array.from({ length: props.pages }).map((_, i) => {
```